test/models/secret_storage_test.rb

```diff
@@ -91,4 +91,44 @@
       end
     end
   end
+
+  describe SecretStorage::HashicorpVault do
+    # note, we need to call the storage engine here directly
+    # as the model has already loaded it's config
+    # from ENV
+    describe ".read" do
+      setup do
+        data = {foo: 'bar', data: { secret_data:"bar"}}.to_json
+        stub_request(:get, "https://127.0.0.1:8200/v1/secret%2Ffoo%252Fisbar").
+          to_return(status: 200, body: data, headers: {'Content-Type': 'application/json'})
+      end
+
+      it "gets a value based on a key with /s" do
+        SecretStorage::HashicorpVault.read('foo/isbar').must_equal('bar')
+      end
+    end
+
+    describe ".delete" do
+      setup do
+        stub_request(:delete, "https://127.0.0.1:8200/v1/secret%2Ffoo%2Fisbar").
+          to_return(status: 200, body: '', headers: {'Content-Type': 'application/json'})
+      end
+
+      it "deletes a key based on a key with /s" do
```

Comment:
that's great :/

On Thu, Apr 21, 2016 at 11:22 AM, ian waters <notifications@github.com>
wrote:

> In test/models/secret_storage_test.rb
> <https://github.com/zendesk/samson/pull/862#discussion_r60630930>:
>
> > +        stub_request(:get, "https://127.0.0.1:8200/v1/secret%2Ffoo%252Fisbar").
> > +          to_return(status: 200, body: data, headers: {'Content-Type': 'application/json'})
> > +      end
> > +
> > +      it "gets a value based on a key with /s" do
> > +        SecretStorage::HashicorpVault.read('foo/isbar').must_equal('bar')
> > +      end
> > +    end
> > +
> > +    describe ".delete" do
> > +      setup do
> > +        stub_request(:delete, "https://127.0.0.1:8200/v1/secret%2Ffoo%2Fisbar").
> > +          to_return(status: 200, body: '', headers: {'Content-Type': 'application/json'})
> > +      end
> > +
> > +      it "deletes a key based on a key with /s" do
>
> client/api returns true even when deleting a non-existent key.
> [4] pry(main)> Vault.logical.delete('secret/nowaynohowman')
> => true
>
> —
> You are receiving this because you are on a team that was mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/zendesk/samson/pull/862/files/2180fdc1cb5d01dbf925d1a4df3d40ad67d525ee#r60630930>
>